src/test/java/it/pagopa/transactions/services/v1/TransactionServiceTests.java

```diff
@@ -1154,6 +1154,43 @@ void shouldThrowsInvalidRequestExceptionForInvalidClientID() {
         assertThrows(InvalidRequestException.class, () -> transactionsServiceV1.convertClientId(clientId.name()));
     }
 
+    @Test
+    void shouldThrowPaymentMethodNotFoundExceptionForPaymentMethodNotFound() {
+        Transaction transaction = TransactionTestUtils.transactionDocument(
+                it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto.ACTIVATED,
+                ZonedDateTime.now()
+        );
+
+        RequestAuthorizationRequestDto authorizationRequest = new RequestAuthorizationRequestDto()
+                .amount(0)
+                .paymentInstrumentId("paymentInstrumentId")
+                .language(RequestAuthorizationRequestDto.LanguageEnum.IT)
+                .fee(0)
+                .pspId("PSP_CODE")
+                .isAllCCP(false)
+                .details(new ApmAuthRequestDetailsDto().detailType("apm"));
+
+        /* preconditions */
+
+        PaymentMethodNotFoundException exception = new PaymentMethodNotFoundException(
+                UUID.randomUUID().toString(),
+                "CHECKOUT"
+        );
+
+        Mockito.when(ecommercePaymentMethodsClient.getPaymentMethod(any(), any())).thenReturn(Mono.error(exception));
+
+        Mockito.when(repository.findById(TRANSACTION_ID))
+                .thenReturn(Mono.just(transaction));
+
+        /* test */
+
+        assertThrows(
+                PaymentMethodNotFoundException.class,
+                () -> transactionsServiceV1
+                        .requestTransactionAuthorization(TRANSACTION_ID, null, authorizationRequest).block()
+        );
```

Comment:
```suggestion
         StepVerifier.create(
                transactionsServiceV1
                        .requestTransactionAuthorization(TRANSACTION_ID, null, authorizationRequest)
        )
                .expectError(PaymentMethodNotFoundException.class)
                .verify();
```
use StepVerifier instead of block()